File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_section_image_response.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WelcomeManualSectionImageResponse(BaseModel):
    id: uuid.UUID
    section_id: uuid.UUID
    storage_key: str
    caption: str | None = None
    display_order: int
    created_at: datetime
    # Short-lived signed URL minted per request via the single presigned seam.
    presigned_url: str | None = None
    # ``False`` means the underlying MinIO object is missing — the UI renders a
    # placeholder + "re-upload" affordance instead of a broken image.
    is_available: bool = True

    model_config = ConfigDict(from_attributes=True)

File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-section-image-response.ts
/**
 * Mirrors backend `WelcomeManualSectionImageResponse`. ``presigned_url`` is a
 * short-lived signed URL minted per request; ``is_available === false`` means
 * the underlying object is missing and the UI renders a placeholder.
 */
export interface WelcomeManualSectionImageResponse {
  id: string;
  section_id: string;
  storage_key: string;
  caption: string | null;
  display_order: number;
  created_at: string;
  presigned_url?: string | null;
  is_available?: boolean;
}
